resteasy-core/src/main/java/org/jboss/resteasy/plugins/interceptors/CacheControlFeature.java

```diff
@@ -74,4 +74,20 @@ protected CacheControl initCacheControl(NoCache value) {
             cacheControl.getNoCacheFields().add(field);
         return cacheControl;
     }
+
+    private Method findInterfaceBasedMethod(Class<?> root, Method method) {
+        if (method.getDeclaringClass().isInterface() || root.isInterface())
+            return method;
+
+        for (Class<?> intf : root.getInterfaces()) {
+            try {
+                return intf.getMethod(method.getName(), method.getParameterTypes());
```

Comment:
We need to look at the call stack here and determine whether or not this should be a privileged action if the security manager is being used.

Reply:
Just to add, I have used the same approach as for example `@GET` annotation retrieves inheritance. Might be useful to update that as well then.